src/Extensions/AzureBlobPayloads/Converters/LargePayloadDataConverter.cs

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Text;
+
+namespace Microsoft.DurableTask.Converters;
+
+/// <summary>
+/// A DataConverter that wraps another DataConverter and externalizes payloads larger than a configured threshold.
+/// It uploads large payloads to an <see cref="IPayloadStore"/> and returns a reference token string.
+/// On deserialization, it resolves tokens and feeds the underlying converter the original content.
+/// </summary>
+/// <remarks>
+/// Initializes a new instance of the <see cref="LargePayloadDataConverter"/> class.
+/// </remarks>
+/// <param name="innerConverter">The inner data converter to wrap.</param>
+/// <param name="payloadStore">The external payload store to use.</param>
+/// <param name="largePayloadStorageOptions">The options for the externalizing data converter.</param>
+/// <exception cref="ArgumentNullException">Thrown when <paramref name="innerConverter"/>, <paramref name="payloadStore"/>, or <paramref name="largePayloadStorageOptions"/> is null.</exception>
+public sealed class LargePayloadDataConverter(DataConverter innerConverter, IPayloadStore payloadStore, LargePayloadStorageOptions largePayloadStorageOptions) : DataConverter
+{
+    const string TokenPrefix = "blob:v1:"; // matches BlobExternalPayloadStore
```

Comment:
It's problematic from a design perspective that `LargePayloadDataConverter` has direct knowledge of `BlobPayloadStore`. Only the `BlobPayloadStore` should know about the token format, especially since it has `blob` in the name. Consider adding a method to the `IPayloadStore` abstraction, like `IsKnownPayloadToken(string value)` which this class can call into on line 73 to keep the abstraction clean.